pycqed/analysis_v2/coherence_analysis.py

```diff
@@ -1,27 +1,302 @@
 '''
 Hacked together by Rene Vollmer
+Cleaned up (a bit) by Adriaan
 '''
-
+import os
+import matplotlib.pyplot as plt
 import datetime
+from collections import OrderedDict
+from copy import deepcopy
 import pycqed.analysis_v2.base_analysis as ba
-from pycqed.analysis_v2.base_analysis import plot_scatter_errorbar_fit, plot_scatter_errorbar
-from pycqed.analysis import measurement_analysis as ma_old
+from pycqed.analysis_v2.base_analysis import plot_scatter_errorbar_fit,\
+    plot_scatter_errorbar, set_xlabel, set_ylabel
 
 import numpy as np
 import lmfit
+from pycqed.analysis.tools.plotting import SI_val_to_msg_str, \
+    format_lmfit_par, plot_lmfit_res
 
 from pycqed.analysis import analysis_toolbox as a_tools
 
 
+class CoherenceAnalysis(ba.BaseDataAnalysis):
+    """
+    Power spectral density analysis of transmon coherence.
+
+    Note, analysis of the coherence times is separated from data extraction
+    as that can be a complicated process that is highly experiment dependent.
+
+    Args:
+        coherence_table : table containing the data, see below for
+            specification.
+        freq_resonator: readout resonator frequency (in Hz)
+        Qc:             coupling Q of the readout resonator
+        chi_shift       in Hz
+        savename:       figures are saved in a folder created based on the
+            provided t_stop in the datadir. `savename` is used to name
+            this folder.
+
+
+    Coherence_table specification:
+           Row  | Content (arrays)
+        --------+--------
+            1   | dac
+            2   | frequency (Hz)
+            3   | T1 (s)
+            4   | T2 star (s)
+            5   | T2 echo (s)
+            6   | Exclusion mask (True where data is to be excluded)
```

Comment:
Figuring out to set to True in the exclusion mask is a huge pain. Additional helper plots would be great. Eg. one with that use the an index as a marker for the corresponding point. Just make sure dpi is large enough to make it readable. Or use vector format.

Reply:
I think that is a good addition.